ocssw tutorials

aewindle110 opened the pull request:
@itcarroll, if you have time could you take a look at this draft l2gen tutorial? There are some TODOs I've left throughout the notebook that I'd appreciate your help with.

itcarroll commented:
I've only made it up to section 4. Will pick up again tomorrow.

1. I don't know an easy way to speed up the plotting. I have a guess why it is slower than usual (the built-in chunks have 40 bands, which always have to be retrieved even though you only ask for 1). If I'm right, there's nothing we can do.

1. I'm surprised you have to manually put the ancillary files in the par file. It doesn't use those if available, and then default to the common/climatologies?

1. Too bad that the lowest memory server does not work for this notebook! Does that hitch go away without doing uncertainty? Would definitely make it worth it, I think.

1. TIL: you can use `par` twice! No need to copy-pasta anc info.

I made other scattered text edits. Including switching to "par" files from what I would read as "dot par" files.

itcarroll commented:
Completed review with more edits that help couple python variables with par files (avoids pasting in generated strings). I will read through 1 more time today to make sure I didn't break or clutter anything.

Whats the difference between `Rrs` and `Rrs_vvv`? Can you just do `Rrs_550` as the only product (for speed in the tutorial)?

aewindle110 commented:
Awesome, thanks Ian! 

I don't know if there is a difference between Rrs and Rrs_vvv but Rrs_vvv adds each wavelength as a separate variable. I'll check with Jeremy- he included that in his edits.

aewindle110 commented:
Hmmm. Maybe. Maybe it'll go away when we compare with a v3.1 Rrs file?

aewindle110 commented:
Your edits are awesome Ian- thanks for teaching me how to use ocssw programs more efficiently! I’ve deleted all the bold TODO’s within the notebook except the last one about why the chl scatterplot values aren’t different when they should be.. I’ll run this by Jeremy to see if he has thoughts. 

Just have these remaining questions:

- Do we still need to say this since we have a V tag? “This notebook uses OCSSW's new S3 authentication feature, which is still in testing.”
- are you getting “error displaying widget: model not found” when running earthaccess.open?
- The getanc program saved a file as l1b_paths[0].anc for me. Is it just supposed to be l1b_paths.anc?
- Not sure about "pix_sub": 1 and "sc_sub” on l2extract. How did you even find out these were options?
- Still not sure what is causing the ‘oddball points’ in the scatterplot but could see someone asking about this. Maybe Bryan will have an answer…
- Changed “Rrs_vvv” to just “Rrs”
